### PR TITLE
Development setup with demo data

### PR DESCRIPTION
**Pull request by michaelglenister**
## Description
Development setup with demo data

## Related Issue
https://tree.taiga.io/project/jbothma-wazimap-ng/task/629
https://trello.com/c/W3uqNrZM/464-dev-test-data-as-a-developer-i-would-like-a-backend-with-realistic-data-so-that-im-developing-against-the-corner-cases-that-usua

### Added
Created demodata for a quick dev environment setup
## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: demodata.json [hunk 1]
```diff
@@ -0,0 +1,3574 @@
+/usr/local/lib/python3.8/dist-packages/django_q/conf.py:136: UserWarning: Retry and timeout are misconfigured. Set retry larger than timeout, 
+        failure to do so will cause the tasks to be retriggered before completion. 
+        See https://django-q.readthedocs.io/en/latest/configure.html#retry for details.
+  warn("""Retry and timeout are misconfigured. Set retry larger than timeout,
```
Comment (paulmwatson) [inline demodata.json:4]: Python warning/error as text inside JSON file itself. @michaelglenister

File: README.md [hunk 2]
```diff
@@ -47,22 +47,20 @@ Version 0.8 is due soon and will fix bugs that currently don't have workarounds.
...
     
-Run the tests using
+Run the tests
 
     docker-compose run --rm -e DJANGO_CONFIGURATION=Test web pytest /app/tests
-    
-Start the backend using 
 
-    docker-compose up
-
```
Comment (jbothma) [inline README.md:65]: why are we removing this?

File: tests/test_dataload.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from django.test import TestCase
```
Comment (jbothma) [inline tests/test_dataload.py:1]: this is fine, but I believe you should also have something like this:

- https://github.com/OpenUpSA/muni-portal-backend/blob/master/.github/workflows/test.yml#L33
- https://github.com/OpenUpSA/municipal-data/blob/master/.github/workflows/test.yml#L29
the literal command a dev is gonna run based on the documentation

And you can do the same with the shapefile:

- wget the shapefile to the CI filesystem
- run the manage command to load it
- check that it works

Is this idea absurd? is it a bad idea? I get a lot of confidence knowing we'll know when this setup stuff breaks, and that I can always have a running system within minutes after `docker compose down --volumes`

File: README.md (hunk 2 above)
Comment (michaelglenister) [reply README.md:65]: it's moved to line 58, I felt it better to have the data loading and start up commands in one group

File: tests/test_dataload.py (hunk 3 above)
Comment (michaelglenister) [reply tests/test_dataload.py:1]: Added tests for these suggestions. For the shapefile test, I just use the country level - just enough to test the functionality

File: README.md [hunk 4]
```diff
@@ -47,22 +47,20 @@ Version 0.8 is due soon and will fix bugs that currently don't have workarounds.
 
 Local development is normally done inside docker-compose so that the supporting services are available and the environment is very similar to how the application is run in production.
 
-Make docker-compose start the supporting services using
+Make docker-compose start the supporting services
 
     docker-compose run --rm web python wait_for_postgres.py
```
[the remark this answers is not in the document]
Comment (michaelglenister) [reply README.md:52]: The script attempted to connect outside the try-catch, I also extended the timeout. Sometimes Postgres took longer than 30 seconds to start

**Pull request merged**